app/components/preprint-form-upload.js

```diff
@@ -22,6 +46,21 @@ const component = CpPanelBodyComponent.extend(PreprintFormFieldMixin, {
         },
         changeUploadState(newState) {
             this.set('uploadState', newState);
+        },
+        deselectNode() {
+            this.set('selectedNode', null);
+        },
+        selectNodeFile(file) {
+            this.send('selectFile', file.get('links.download'));
+        },
+        selectNode(node) {
+            console.log(node);
+        },
+        selectFile(url) {
+            console.log(url);
```

Comment:
You may want to give some thought to data flow: how will the selection choice be used outside this component, to fill in the overall form responses? Actions (or closure actions) might be helpful.